### PR TITLE
Add error messages for custom validation schemas

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This adds error messages for custom validation schemas.

## Changes

- Reused schemas now have the following error messages:
  - `NumericStringSchema` - 'Invalid base-10 numeric string'
  - `HexSchema` - 'Invalid "0x" notated hex string'
  - `AddressSchema` - 'Invalid address'
- A few isolated fields that have custom validation have had error messages added.
- The above changes have been propegated across test coverage.

### CODE REVIEW

**Review by hectorgomezv — APPROVED** [review]
👏🏻

File: src/validation/entities/schemas/hex.schema.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { z } from 'zod';
 import { isHex } from 'viem';
 
-export const HexSchema = z.string().refine(isHex);
+export const HexSchema = z.string().refine(isHex, {
+  message: 'Invalid hex string',
+});
```
Comment (fmrsabino) [inline src/validation/entities/schemas/hex.schema.ts:6]: The `isHex` requires a string to start with `0x` right? If so, I would add this to the error message as a string like `fff` can be considered an hex string for the clients.

File: src/validation/entities/schemas/numeric-string.schema.ts [hunk 2]
```diff
@@ -4,4 +4,6 @@ function isNumeric(value: unknown): boolean {
   return typeof value === 'string' && !isNaN(Number(value));
 }
 
-export const NumericStringSchema = z.string().refine(isNumeric);
+export const NumericStringSchema = z.string().refine(isNumeric, {
+  message: 'Invalid numeric string',
+});
```
Comment (fmrsabino) [inline src/validation/entities/schemas/numeric-string.schema.ts:9]: Do we consider only 10-base numbers? Or in any base?

File: src/validation/entities/schemas/hex.schema.ts (hunk 1 above)
Comment (iamacook) [reply src/validation/entities/schemas/hex.schema.ts]: Good point. I updated the message to be `'Invalid "0x" notated hex string"` in 30596f4249a57df3c13f8aedd61c025d817dd81b.

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**